stm/stm.ml

```diff
@@ -496,21 +494,14 @@ end = struct (* {{{ *)
 
   let checkout_shallowest_proof_branch () =
     if List.mem edit_branch (Vcs_.branches !vcs) then begin
-      checkout edit_branch;
-      match get_branch edit_branch with
-      | { kind = `Edit (mode, _,_,_,_) } -> Proof_global.activate_proof_mode mode
-      | _ -> assert false
+      checkout edit_branch
     end else
       let pl = proof_nesting () in
       try
-        let branch, mode = match Vcs_aux.find_proof_at_depth !vcs pl with
-          | h, { Vcs_.kind = `Proof (m, _) } -> h, m | _ -> assert false in
-        checkout branch;
-        stm_prerr_endline (fun () -> "mode:" ^ mode);
-        Proof_global.activate_proof_mode mode
+        let branch = fst @@ Vcs_aux.find_proof_at_depth !vcs pl in
```

Comment:
I think this is a kind of little-endian / big-endian problem in Ocaml context.

There is a group of people who avoids these operators entirely.

Then there are people who prefer `@@`.

There is a group of people who [oppose](https://github.com/ocaml/ocaml/pull/791) to usage of, GASP, functional composition.

And then there are people who prefer `|>` and `%>`.
(I belong to this group. I like to see the operations in  the same as pipelines in UNIX).